src/OpenTelemetry.Contrib.Instrumentation.Owin/Implementation/OwinInstrumentationEventSource.cs

```diff
@@ -0,0 +1,94 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Diagnostics.Tracing;
+using System.Globalization;
+using System.Threading;
+
+namespace OpenTelemetry.Contrib.Instrumentation.Owin
+{
+    /// <summary>
+    /// EventSource events emitted from the project.
+    /// </summary>
+    [EventSource(Name = "OpenTelemetry-Instrumentation-Owin")]
+    internal class OwinInstrumentationEventSource : EventSource
+    {
+        public static OwinInstrumentationEventSource Log { get; } = new OwinInstrumentationEventSource();
+
+        [NonEvent]
+        public void RequestFilterException(Exception ex)
+        {
+            if (this.IsEnabled(EventLevel.Error, (EventKeywords)(-1)))
+            {
+                this.RequestFilterException(ToInvariantString(ex));
+            }
+        }
+
+        [Event(EventIds.RequestIsFilteredOut, Message = "Request is filtered out.", Level = EventLevel.Verbose)]
+        public void RequestIsFilteredOut()
+        {
+            this.WriteEvent(EventIds.RequestIsFilteredOut);
+        }
+
+        [Event(EventIds.RequestFilterException, Message = "InstrumentationFilter threw exception. Request will not be collected. Exception {0}.", Level = EventLevel.Error)]
+        public void RequestFilterException(string exception)
+        {
+            this.WriteEvent(EventIds.RequestFilterException, exception);
+        }
+
+        [NonEvent]
+        public void EnrichmentException(Exception exception)
+        {
+            if (this.IsEnabled(EventLevel.Error, (EventKeywords)(-1)))
+            {
+                this.EnrichmentException(ToInvariantString(exception));
+            }
+        }
+
+        [Event(EventIds.EnrichmentException, Message = "Enrichment threw exception. Exception {0}.", Level = EventLevel.Error)]
+        public void EnrichmentException(string exception)
+        {
+            this.WriteEvent(EventIds.EnrichmentException, exception);
+        }
+
+        /// <summary>
+        /// Returns a culture-independent string representation of the given <paramref name="exception"/> object,
+        /// appropriate for diagnostics tracing.
+        /// </summary>
+        private static string ToInvariantString(Exception exception)
+        {
+            var originalUICulture = Thread.CurrentThread.CurrentUICulture;
+
+            try
+            {
+                Thread.CurrentThread.CurrentUICulture = CultureInfo.InvariantCulture;
+                return exception.ToString();
+            }
+            finally
+            {
+                Thread.CurrentThread.CurrentUICulture = originalUICulture;
+            }
+        }
+
+        private class EventIds
+        {
+            public const int RequestIsFilteredOut = 1;
+            public const int RequestFilterException = 2;
+            public const int EnrichmentException = 3;
+        }
+    }
```

Comment:
I like this pattern

Reply:
Credit to @mconnew https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/38#discussion_r577945650